### PR TITLE
Require the Bitbucket Server URL, project and repo name to always be passed for `bbs2gh migrate-repo`

### PR DESCRIPTION
**Pull request by timrogers**
This makes a **breaking change** to `bbs2gh migrate-repo` requiring the `--bbs-server-url`, `--bbs-project` and `--bbs-repo` to always be passed, even in the flows where the CLI doesn't talk to the Bitbucket Server API.

This change allows us to always construct the URL of the repository and then send it to the GitHub API when starting the migration. 

That's important, because the URL allows us to intelligently pick the right repo to migrate from the archive, rather than picking the "first" repo in the fork hierarchy (which may be wrong!).

I did consider introducing a warning first and then making a breaking change, but I thought going straight to a breaking change was the right decision because:

* This is currently in private beta, so I can easily contact affected users
* We're getting ready to move to public beta very soon, and I want this to be "right" before the public beta
* The impact of this change is limited, given that most users use the default end-to-end flow

Fixes https://github.com/github/gh-gei/issues/1056.

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/OctoshiftCLI.Tests/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgsTests.cs [hunk 1]
```diff
@@ -283,44 +283,6 @@ public void Errors_If_BbsServer_Url_Not_Provided_But_Smb_User_Is_Provided()
                 .WithMessage("*SSH*SMB*--bbs-server-url*");
         }
 
-        [Fact]
```
Comment (timrogers) [inline src/OctoshiftCLI.Tests/bbs2gh/Commands/MigrateRepo/MigrateRepoCommandArgsTests.cs:286]: We now validate in the `Args` class that all 3 arguments are always provided

**Comment by timrogers** [comment]
I've tested all three flows (end-to-end + upload-and-import + import-only) after this change, and they all behave as expected.

**Comment by timrogers** [comment]
Docs changes in https://github.com/github/docs-internal/pull/38838

**Comment by dylan-smith** [comment]
Did you consider doing a 2-phase implementation (like we have done for other breaking changes). Where we first make those arguments optional when using `--archive-path` or `--archive-url` and spit out a warning if users don't provide the optional arguments.

Then at some point in the future come back and make them required.

**Comment by timrogers** [comment]
> Did you consider doing a 2-phase implementation (like we have done for other breaking changes). Where we first make those arguments optional when using `--archive-path` or `--archive-url` and spit out a warning if users don't provide the optional arguments.
> 
> 
> 
> Then at some point in the future come back and make them required.

I did consider it - but I feel like a more aggressive approach is fair game during a limited private beta that is heading to public beta in the near future. 

It's easy for me to contact affected users, so I thought a more immediate change made sense to fix what is a pretty nasty bug. 

But I will think more on this over the weekend! Thanks for raising it.

**Review by dpmex4527 — APPROVED** [review]
LGTM!

**Review by dpmex4527 — APPROVED** [review]

**Pull request merged**